### PR TITLE
feat: adding XAxis to BigNumberTrend

### PR DESCRIPTION
**Pull request by zhaoyongjie**
### SUMMARY

Apply temporal XAxis to BigNumber with Trendline

### After
image

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset-frontend/packages/superset-ui-chart-controls/src/shared-controls/constants.tsx [hunk 1]
```diff
@@ -51,3 +52,29 @@ export const xAxisControlConfig = {
   },
   default: undefined,
 };
+
+export const temporalXAxisMixin = {
```
Comment (zhaoyongjie) [inline superset-frontend/packages/superset-ui-chart-controls/src/shared-controls/constants.tsx:56]: copy/paste from `dndGranularitySqlaControl`

Comment (villebro) [inline superset-frontend/packages/superset-ui-chart-controls/src/shared-controls/constants.tsx]: As this is a mixin, we maybe the type could be narrowed down to something like `Pick[BaseControlConfig, 'label' | 'mapStateToProps']`?

Also, I'm wondering if we should make this a function `getXAXisControlMixin` that takes an optional argument `dataTypes: GenericDataType[]`, which if defined, would only return the requested column types? For instance, at some point in the future I'm assuming we want to support any ordinal or linear datatypes in this chart type, and in that case we would likely want to support `GenericDataType.NUMERIC` and `GenericDataType.TEMPORAL`.

**Review by villebro — APPROVED** [review]
LGTM and works as expected with a minor typing related comment. I know this is out of scope for this PR, but I'd love to see us add support for at least numerical types on the x-axis.

File: superset-frontend/packages/superset-ui-chart-controls/src/shared-controls/constants.tsx (hunk 1 above)
Comment (zhaoyongjie) [reply superset-frontend/packages/superset-ui-chart-controls/src/shared-controls/constants.tsx]: Thanks for the review and very useful case for this mixin. I changed this variable from the `temporalXAxisMixin` to the `temporalColumnMixin` and removed `mapStateToProps` from the original one from `dndGranularitySqlaControl`. I totally agree with making a function for returning specific column types from the original datasource. I will refactor this part in separate PR.

**Pull request merged**